unitTests/ui/OmniboxAnalyticsTest.ts

```diff
@@ -0,0 +1,102 @@
+import * as Mock from '../MockEnvironment';
+import { Omnibox } from '../../src/ui/Omnibox/Omnibox';
+import { IOmniboxAnalytics } from '../../src/ui/Omnibox/OmniboxAnalytics';
+import { IAnalyticsOmniboxSuggestionMeta, analyticsActionCauseList } from '../../src/ui/Analytics/AnalyticsActionListMeta';
+import { Suggestion } from '../../src/magicbox/SuggestionsManager';
+
+export function OmniboxAnalyticsTest() {
+  let omniboxAnalytics: IOmniboxAnalytics;
+
+  describe('omniboxAnalyticsTest', () => {
+    let testEnv: Mock.MockEnvironmentBuilder;
+
+    function setupEnv() {
+      testEnv = new Mock.MockEnvironmentBuilder();
+      testEnv.searchInterface.getOmniboxAnalytics = jasmine.createSpy('omniboxAnalytics').and.returnValue(setupOmniboxAnalytics()) as any;
+    }
+
+    function setupOmniboxAnalytics(): IOmniboxAnalytics {
+      const partialQueries: string[] = [];
+      let suggestionRanking: number;
+      const suggestions: string[] = [];
+      let partialQuery: string;
+      const buildCustomDataForPartialQueries = (): IAnalyticsOmniboxSuggestionMeta => {
+        const partialQueries = omniboxAnalytics.partialQueries.join(';');
+        const suggestions = omniboxAnalytics.suggestions.join(';');
+
+        return {
+          partialQueries,
+          suggestionRanking: omniboxAnalytics.suggestionRanking,
+          suggestions,
+          partialQuery: omniboxAnalytics.partialQuery
+        };
+      };
+      return (omniboxAnalytics = {
```

Comment:
Since the method has `setup` in the name, I would not return anything. If someone needs the value, they can access it from the variable `omniboxAnalytics`.